src/Config.php

```diff
@@ -29,21 +29,23 @@ class Config
     private $gcp_call_invoker;
 
     /**
-     * @param string  $hostname Which API we want to manage the connection.
+     * @param string  $target The target API we want to manage the connection.
      * @param \Grpc\Gcp\ApiConfig   $conf
      * @param CacheItemPoolInterface $cacheItemPool A pool for storing configuration and channels
      *                                            cross requests within a single worker process.
      */
-    public function __construct($hostname, $conf = null, CacheItemPoolInterface $cacheItemPool = null)
+    public function __construct($target, $conf = null, CacheItemPoolInterface $cacheItemPool = null)
     {
         if ($conf == null) {
             // If there is no configure file, use the default gRPC channel.
             $this->gcp_call_invoker = new \Grpc\DefaultCallInvoker();
             return;
         }
         $gcp_channel = null;
-        $this->hostname = $hostname;
-        $channel_pool_key = $hostname . 'gcp_channel' . getmypid();
+        $url_host = parse_url($target, PHP_URL_HOST);
+        $this->hostname = $url_host ? $url_host : $target;
+        $channel_pool_key = $this->hostname . '.gcp.channel.' . getmypid();
```

Comment:
Line 45-46 are for better key names (in case target are passed in with port number, e.g. 'spanner.googleapis.com:443' )